File: lib/commands/builtin/forward.ts
import type { CommandSpec } from "../types"

export const forwardCommand: CommandSpec = {
  name: "forward",
  aliases: ["fwd"],
  summary: "navigate the tab history forward",
  usage: ["forward [tabId]", "fwd [tabId]"],
  man: [
    "NAME",
    "  forward, fwd - navigate the tab history forward",
    "",
    "SYNOPSIS",
    "  forward [tabId]",
    "  fwd [tabId]"
  ],
  async execute(ctx, args) {
    const tab = await ctx.resolveTabArg(args[1])
    if (!tab?.id) {
      return ["no target tab (set focus window or pass tabId)"]
    }
    await chrome.tabs.goForward(tab.id)
    return [`goForward tab ${tab.id}`]
  }
}

File: wasm/bmxt-core/src/cmd/forward.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "forward",
    aliases: &["fwd"],
    usage_primary: "forward [tabId]",
    man: &[
        "NAME",
        "  forward, fwd - navigate the tab history forward",
        "",
        "SYNOPSIS",
        "  forward [tabId]",
        "  fwd [tabId]",
    ],
};

pub fn run(args: &[String]) -> DispatchJson {
    let tab_arg = args.get(1).cloned();
    DispatchJson::effects(vec![Effect::GoForward { tab_id_arg: tab_arg }])
}
